module/spl/spl-kmem.c

```diff
@@ -133,6 +133,32 @@ strfree(char *str)
 }
 EXPORT_SYMBOL(strfree);
 
+#ifndef __GFP_REPEAT
+#define	__GFP_REPEAT	__GFP_RETRY_MAYFAIL
+#endif
+void *
+spl_kvmalloc(size_t size, gfp_t flags)
+{
+	gfp_t kmalloc_flags = flags;
+	void *ret;
+#ifdef HAVE_KVMALLOC
+	if ((flags & GFP_KERNEL) == GFP_KERNEL) {
+		return (kvmalloc(size, flags));
+	}
+#endif
+	if (size > PAGE_SIZE) {
+		kmalloc_flags |= __GFP_NOWARN;
+		if (!(kmalloc_flags & __GFP_REPEAT) ||
+		    (size <= PAGE_SIZE << PAGE_ALLOC_COSTLY_ORDER)) {
+			kmalloc_flags |= __GFP_NORETRY;
+		}
+	}
+	ret = kmalloc_node(size, kmalloc_flags, NUMA_NO_NODE);
```

Comment:
`spl_kmem_alloc_impl` declares `ptr` at the top which is, I assume, the reason fo this here

Reply:
We didn't always allow c99, so there's a mix of styles depending on when a function was written.   At the top would be fine here too since it's a small function.

Reply:
To keep it consistent with all the other functions ins spl-kmem.c I keep it at the top